dmoj/cptbox/isolate.py

```diff
@@ -44,11 +45,24 @@ def __init__(self, read_fs, write_fs=None, writable=(1, 2)):
             self._getcwd_pid = lambda pid: os.readlink('/proc/%d/cwd' % pid)
             self._getfd_pid = lambda pid, fd: os.readlink('/proc/%d/fd/%d' % (pid, fd))
 
+        if has_landlock():
+            if debug_landlock:
+                self.update(
+                    {sys_openat: self.debug_file_access_at('openat'), sys_open: self.debug_file_access('open', 0)}
```

Comment:
Are these the only calls landlock restricts? You might want to add more...

Reply:
Landlock doesn't restrict a lot of calls right now. From the [docs](https://www.kernel.org/doc/html/latest/userspace-api/landlock.html#landlock-rules):

> It is currently not possible to restrict some file-related actions accessible through these syscall families: chdir(2), truncate(2), stat(2), flock(2), chmod(2), chown(2), setxattr(2), utime(2), ioctl(2), fcntl(2), access(2). Future Landlock evolutions will enable to restrict them.

This leaves us with... basically nothing. Most of our file access calls are variants on `stat`, `access`, and `readlink`, none of which can be restricted. Notably, `readlink` is not considered a "file-related" action, as far as I understand: landlock hooks in _after_ the resolution is complete, including symlinks.

I wish it were more.

Reply:
I guess, if nothing else; `open` is probably the most important syscall set? I guess I don't know what the syscall counts look like.